src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -148,6 +152,31 @@ public function onKernelController(ControllerEvent $event): void
 
         $request->attributes->set('_mounted_component', $mounted);
 
+        // autowire live file arguments
+        if ($request->files->count()) {
+            $allFiles = $request->files->all();
+            $accessor = PropertyAccess::createPropertyAccessor();
+            foreach (LiveFileArg::liveFileArgs($component, $action) as $parameter => $fileArg) {
+                $path = $fileArg->getPropertyPath();
+
+                if (!$accessor->isReadable($allFiles, $path)) {
+                    throw new \RuntimeException(sprintf('File path "%s" for parameter "%s" is invalid', $fileArg->name, $parameter));
+                }
+
+                if ($files = $accessor->getValue($allFiles, $path)) {
+                    $request->attributes->set(
+                        $parameter,
+                        $fileArg->multiple ? $files : $files[0]
```

Comment:
Do we need `$fileArg->multiple` at all? If the `<input file` has `multiple`, then `$files` will be an array of `UploadedFile`. If it is NOT `multiple`, then it will already be a single `UploadedFile`. I don't think we need to do anything.

Reply:
This was my thought as well but do we need to be worried about someone fudging the `<file>` input to add `multiple` then uploading multiple files?

What about checking the typehint? `array === multiple`, `UploadedFile === single`.

Reply:
Hmm. It we do nothing, and someone messes with things, it would explode with a 500 error (since an array would fail to be passed to an arg typed with `UploadedFile`). So, I don't think there's an attack vector. However, allowing bad users to trigger a 500 isn't too great either ;). 

So, yea, we COULD check the uploaded data first to see if we're dealing with ONE or MULTIPLE files. Then we COULD check the type on the arg to see if it's something compatible (e.g. no type, or `UploadedFile` for a single, etc). If it is NOT compatible, we return a 400 error.

That DOES feel a little magic... I can't think of any similar thing that's done in Symfony. Like, is there a way in php to basically ask "Hey: would this value be compatible with this argument type?". WDYT?

Reply:
I don't think there is a simple, PHP way of doing it. And I agree removing `multiple` on `fileArg` could be beneficial. After all we have typehints. I will refactor this part for sure :+1: